crates/db_schema/src/traits.rs

```diff
@@ -140,11 +140,6 @@ pub trait Reportable {
     Self: Sized;
 }
 
-// TODO these should be removed, there should be another way to do this
-pub trait DeleteableOrRemoveable {
-  fn blank_out_deleted_or_removed_info(self) -> Self;
-}
```

Comment:
So the reason this isnt needed anymore is because normal users dont see removed content at all now, right? I think this will require a change in lemmy-ui so that removed content isnt rendered unless you click a button. Otherwise if there are offensive images or text posted, it would be difficult to hide.

![Screenshot_20230224_181239](https://user-images.githubusercontent.com/1044450/221139098-67777400-13ad-4670-97ec-8b0949a1795d.png)

Something like this in phpBB, you have to click a button to see the content of a deleted post.

Reply:
That's correct for posts, but they shouldn't come back at all, for anyone but moderators / admins and creators. The front ends could still blur / hide content similar to what's above tho (at least for the mods and admins).

For comments, all results **do need** to come back, because otherwise the comment tree could be missing some branches.  The front ends use the `deleted` and removed fields to hide results anyway, so this blanking was pretty pointless and needlessly complicated.